backend/tests/test_storage_service.py

```
@@ -137,6 +139,130 @@ def test_surfaces_supabase_body_on_failure(self):
+    # called synchronously, so a MagicMock side effect is correct.
+    cls_mock = MagicMock(return_value=client_instance)
+    return patch("services.storage_service.httpx.AsyncClient", cls_mock), client_instance
+
+
+class TestEnsureBucketExists:
+    """Pin the startup bootstrap behavior. Backend lifespan calls this
+    on app boot so new Supabase environments self-create the avatars
+    bucket — fixes the underlying cause of issue #75 (bucket missing,
+    every avatar upload returned 502 'Bucket not found')."""
+
+    KW = dict(public=True, file_size_limit=5_242_880, allowed_mime_types=["image/png"])
```

Ruff (0.15.12)

[warning] Mutable default value for class attribute

(RUF012)